src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/armor/RadiationTask.java

```diff
@@ -74,4 +86,39 @@ protected void onPlayerTick(Player p, PlayerProfile profile) {
             }
         }
     }
+
+    /**
+     * Checks if the {@link Player} is within their grace period. A grace period is granted after death
+     * to give enough time to remove the radioactive items before being killed once more, which
+     * with KeepInventory on would result in a very hard-to-escape loop.
+     * @see io.github.thebusybiscuit.slimefun4.implementation.listeners.RadioactivityListener
+     *
+     * @param player
+     *              The {@link Player} to check against.
+     *
+     * @return Returns true if the {@link Player} is within their grace period.
+     */
+    private boolean withinGracePeriod(@Nonnull Player player) {
+        long gracePeriodEnd = ACTIVE_GRACE_PERIODS.getOrDefault(player.getUniqueId(), 0L);
+        if (gracePeriodEnd >= System.currentTimeMillis()) {
+            // Player is within their grace period
+            return true;
+        }
+
+        if (gracePeriodEnd > 0) {
```

Comment:
if its 0 its lapsed as well

Reply:
If it's 0 then it's not in the map so no need to remove